components/script/dom/element.rs

```diff
@@ -2120,6 +2121,59 @@ impl Element {
         node.owner_doc().element_attr_will_change(self, attr);
     }
 
+    /// <https://html.spec.whatwg.org/multipage/#the-style-attribute>
+    fn update_style_attribute(&self, attr: &Attr, mutation: AttributeMutation) {
+        let doc = self.upcast::<Node>().owner_doc();
+        let source = &**attr.value();
+        // However, if the Should element's inline behavior be blocked by
+        // Content Security Policy? algorithm returns "Blocked" when executed
+        // upon the attribute's element, "style attribute", and the attribute's value,
+        // then the style rules defined in the attribute's value must not be applied to the element. [CSP]
+        if doc.should_elements_inline_type_behavior_be_blocked(
+            self,
+            csp::InlineCheckType::StyleAttribute,
+            source,
```

Comment:
This source value is incorrect. It gets compared against the policy source values like `'self'`, so every inline style change is blocked by CSP checks right now.

Reply:
Sorry, I did some more reading and realized that I was misunderstanding what `style-src 'self'` is supposed to do, and that the spec text describes exactly what the code is doing.